larky/src/main/java/com/verygood/security/larky/modules/vgs/cerebro/text/analyzer/dto/TextPIIEntity.java

```diff
@@ -28,13 +27,17 @@
   // Where the PII ends
   private final StarlarkInt end;
 
+
+  TextPIIEntity(String entityType, StarlarkFloat score, StarlarkInt start, StarlarkInt end, StarlarkThread thread) {
+    super(dictOf(entityType, score, start, end), thread);
+    this.entityType = entityType;
+    this.score = score;
+    this.start = start;
+    this.end = end;
+  }
+
   public static TextPIIEntity of(String entityType, double score, int start, int end) {
```

Comment:
- This would be `immutableOf` if you're going to be passing `null` for the `StarlarkThread`. 
- `of()` should then take `StarlarkThread` as a parameter

Reply:
updated